packages/dioxus-blitz/src/window.rs

```diff
@@ -15,6 +16,57 @@ use winit::event::{ElementState, MouseButton};
 use winit::event_loop::{ActiveEventLoop, EventLoopProxy};
 use winit::{event::WindowEvent, keyboard::KeyCode, keyboard::ModifiersState, window::Window};
 
+struct State {
+    /// Accessibility adapter for `accesskit`.
+    #[cfg(feature = "accesskit")]
+    adapter: accesskit_winit::Adapter,
+
+    #[cfg(feature = "accesskit")]
+    node_ids: std::collections::HashMap<accesskit::NodeId, usize>,
+
+    #[cfg(feature = "accesskit")]
+    id_to_node_ids: std::collections::HashMap<usize, accesskit::NodeId>,
+
+    #[cfg(feature = "accesskit")]
+    next_id: u64,
+
+    /// Main menu bar of this view's window.
+    _menu: Menu,
+}
+
+impl State {
+    fn build_node(
+        &mut self,
+        node_id: usize,
+        node: &Node,
+    ) -> (accesskit::NodeId, accesskit::NodeBuilder) {
+        let mut node_builder = accesskit::NodeBuilder::default();
+        if let Some(element_data) = node.element_data() {
+            let name = element_data.name.local.to_string();
+            let role = match &*name {
+                "button" => accesskit::Role::Button,
+                "div" | "section" => accesskit::Role::Group,
+                "p" => accesskit::Role::Paragraph,
+                _ => accesskit::Role::Unknown,
+            };
+
+            node_builder.set_role(role);
+            node_builder.set_name(name);
+        } else if node.is_text_node() {
+            node_builder.set_role(accesskit::Role::StaticText);
+            node_builder.set_name(node.text_content());
+        }
+
+        let id = accesskit::NodeId(self.next_id);
```

Comment:
Oh... we get to choose the AccessKit ids... Is there any reason not to use the dom's NodeId directly?

Reply:
If we are able to map 1-1 from nodes (and text nodes) I'd say we should totally do this 👀